token/jwt/claims_jwt.go

```diff
@@ -240,6 +225,17 @@ func (c *JWTClaims) FromMap(m map[string]interface{}) {
 	}
 }
 
+func toTime(v interface{}, def time.Time) (t time.Time) {
+	t = def
+	switch a := v.(type) {
+	case float64:
+		t = time.Unix(int64(a), 0).UTC()
```

Comment:
Do we want to preserve sub-seconds if they are ever passed? Like:

```
	secs := int64(1)
	nsecs := int64((1 - float64(secs)) * 1e9)
	t := time.Unix(secs, nsecs).UTC()
```

Reply:
although it like the idea [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519) states that core timestamp claims `exp` and `iat` MUST be given in seconds:

> The "exp" (expiration time) claim ... Its value MUST be a number
   containing a NumericDate value.  Use of this claim is OPTIONAL.

> The "iat" (issued at) claim  ... MUST be a number containing a NumericDate value.  Use of this
   claim is OPTIONAL.
   
> NumericDate
      A JSON numeric value representing the number of seconds from
      1970-01-01T00:00:00Z UTC until the specified UTC date/time,
      ignoring leap seconds. 
      
besides that, it requires some state to make `ToMap` and `FromMap` consistent/equivalent. Once the `time.Time` is created from float64 with sub-seconds, the inverse conversion (`ToMap`) must also contains sub-seconds, which means using a float64 instead of a int64 (which is contrary to what we are doing in this PR :) ), but currently there is no way to infer this, so it will required a flag for that and also we will keep using float64 for some cases. 

it looks nice and clever, but at the end makes things more complicated, and not RFC compliant.

Reply:
Agreed @narg95 !